tests/test_lab/test_states.py

```diff
@@ -191,3 +210,28 @@ def test_getitem_set_modes(modes):
     state2 = State(ket=ket, modes=modes)
 
     assert state1.modes == state2.modes
+
+
+@pytest.mark.parametrize("pure", [True, False])
+def test_concat_pure_states(pure):
+    """Test that fock states concatenate correctly and are separable"""
+    state1 = Fock(1, cutoffs=[15])
+    state2 = Fock(4, cutoffs=[15])
+
+    if not pure:
+        state1 >>= Attenuator(transmissivity=0.95)
+        state2 >>= Attenuator(transmissivity=0.9)
+
+    psi = state1 & state2
+
+    # test concatenated state
+    psi_dm = math.transpose(math.tensordot(state1.dm(), state2.dm(), [[], []]), [0, 2, 1, 3])
+    assert np.allclose(psi.dm(), psi_dm, rtol=1e-2)
```

Comment:
is the high relative tolerance necessary?

Reply:
Not actually. I was testing something and forgot to update this. Thanks for spotting it!